article/helpers.py

```diff
@@ -8,25 +9,93 @@
 from . import structure
 
 
+WORDS_PER_MINUTE = 180  # Average WPM on screen
+
+
 def markdown_to_html(markdown_file_path):
     html = render_to_string(markdown_file_path)
     return markdown2.markdown(html)
 
 
+def filter_lines(lines_list):
+    """BeautifulSoup returns \n as lines as well, we filter them out.
+
+    It's a function because more filtering can be added later.
+    """
+    return filter(lambda x: x != '\n', lines_list)
+
+
+def lines_list_from_html(html):
+    """Parses the HTML to return text lines."""
+    soup = BeautifulSoup(html, 'html.parser')
+    return soup.findAll(text=True)
+
+
+def count_average_word_number_in_lines_list(lines_list, word_length=5):
+    """Assume average word length, counts how many words in all the lines."""
+    total_words = 0
+    for line in lines_list:
+        total_words += len(line)/word_length
+    return total_words
+
+
+def time_to_read_in_minutes(article):
+    """Return rounded time to read in minutes give an Article object."""
+    html = markdown_to_html(article.markdown_file_path)
+    lines_list = lines_list_from_html(html)
+    filtered_lines_list = filter_lines(lines_list)
+    total_words_count = count_average_word_number_in_lines_list(
+        filtered_lines_list
+    )
+    return round(total_words_count / WORDS_PER_MINUTE)
+
+
+def total_time_to_read_multiple_articles(articles):
+    return sum((time_to_read_in_minutes(article) for article in articles))
+
+
 class BaseArticleReadManager(abc.ABC):
     def __init__(self, request):
         self.request = request
 
     persist_article = abc.abstractproperty()
     retrieve_articles = abc.abstractproperty()
 
-    def article_read_count(self, group):
-        read_articles = frozenset(self.retrieve_articles())
-        articles_in_group = structure.ALL_GROUPS_ARTICLES_SETS[group]
+    @staticmethod
+    def articles_group(group_key):
```

Comment:
not `article_group`?

Reply:
It's a group of articles, should it not be plural? I ask the help of @richtier

Reply:
I would not have this method on the `BaseArticleReadManager` - it's just getting from a dictionary `structure.ALL_GROUPS_DICT`

but to answer the question - it gets a single `ArticleGroup` from `ALL_GROUPS_DICT`, so I would call it `get_article_group`.

Reply:
Actually, the same function already exists in `structure.py`. I'll delete this one!